rules/CodeQuality/Rector/Foreach_/SimplifyForeachToArrayFilterRector.php

```diff
@@ -119,6 +127,30 @@ private function shouldSkip(Foreach_ $foreach): bool
         return $ifNode->elseifs !== [];
     }
 
+    private function shouldSkipForeachKeyUsage(If_ $ifNode, Expr $keyVar): bool
+    {
+        if (! $keyVar instanceof Variable) {
+            return false;
+        }
+
+        $keyVarUsage = $this->betterNodeFinder->find(
+            $ifNode,
+            fn (Node $node): bool => $this->exprUsedInNodeAnalyzer->isUsed($node, $keyVar)
+        );
+
+        $keyVarUsageCount = count($keyVarUsage);
+        if ($keyVarUsageCount === 1) {
+            $parentArrayDimFetch = $keyVarUsage[0]->getAttribute(AttributeKey::PARENT_NODE);
```

Comment:
`[0]` key usage may be miss re-index, using `current($keyVarUsage)` will be safer.